pkg/managed/deployment_test.go

```diff
@@ -0,0 +1,171 @@
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package managed
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"github.com/verrazzano/verrazzano-operator/pkg/constants"
+	"github.com/verrazzano/verrazzano-operator/pkg/monitoring"
+	"github.com/verrazzano/verrazzano-operator/pkg/testutil"
+	"github.com/verrazzano/verrazzano-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"os"
+	"reflect"
+	"testing"
+)
+
+// TestCreateDeployments tests the creation of deployments.
+// GIVEN a cluster which does not have any deployments
+//  WHEN I call CreateDeployments
+//  THEN there should be an expected set of deployments created
+func TestCreateDeployments(t *testing.T) {
+	assert := assert.New(t)
+
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+	manifest := testutil.GetManifest()
+	vmiSecret := monitoring.NewVmiSecret(modelBindingPair.Binding)
+	secrets := &testutil.FakeSecrets{Secrets: map[string]*corev1.Secret{
+		constants.VmiSecretName: vmiSecret,
+	}}
+
+	// set env vars needed for test; set them back when the test completes
+	defer setEnv(t, "WLS_MICRO_REQUEST_MEMORY", os.Getenv("WLS_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "WLS_MICRO_REQUEST_MEMORY", "2.5Gi")
+	defer setEnv(t, "COH_MICRO_REQUEST_MEMORY", os.Getenv("COH_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "COH_MICRO_REQUEST_MEMORY", "2.5Gi")
+	defer setEnv(t, "HELIDON_MICRO_REQUEST_MEMORY", os.Getenv("HELIDON_MICRO_REQUEST_MEMORY"))
+	setEnv(t, "HELIDON_MICRO_REQUEST_MEMORY", "2.5Gi")
+
+	err := CreateDeployments(modelBindingPair, clusterConnections, &manifest, "testURI", secrets)
+	assert.Nil(err, "got an error from CreateDeployments: %v", err)
+
+	assertDeployments(err, clusterConnection, assert)
+}
+
+// TestCreateDeployments tests that an existing deployment is properly updated.
+// GIVEN a cluster which has an existing expected deployment (verrazzano-wko-operator)
+//  WHEN I call CreateDeployments
+//  THEN there should be an expected set of deployments created
+//    AND the existing deployment (verrazzano-wko-operator) should be updated as expected
+func TestCreateDeploymentsUpdateExisting(t *testing.T) {
+	assert := assert.New(t)
+
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+	manifest := testutil.GetManifest()
+	vmiSecret := monitoring.NewVmiSecret(modelBindingPair.Binding)
+	secrets := &testutil.FakeSecrets{Secrets: map[string]*corev1.Secret{
+		constants.VmiSecretName: vmiSecret,
+	}}
+
+	// set env vars needed for test; set them back when the test completes
+	defer setEnv(t, "WLS_MICRO_REQUEST_MEMORY", os.Getenv("WLS_MICRO_REQUEST_MEMORY"))
```

Comment:
I had a similar situation where I had to set env vars in util/apiserver/cors_test.go. Mark pointed out it's possible for these tests to potentially run in parallel with the same os environment, so I changed env.go to allow an override Getenv function and used that instead of os.Setenv. You might want to do something similar here. 
https://github.com/verrazzano/verrazzano-operator/blob/master/pkg/util/apiserver/cors_test.go#L165

Reply:
@desagar Thanks for reviewing.  Good point.  I'll make the change.

Reply:
@desagar I made the changes.  Pipeline passed.